desktop_version/src/Game.cpp

```diff
@@ -7486,15 +7457,31 @@ void Game::createmenu( enum Menu::MenuName t, bool samemenu/*= false*/ )
     case Menu::timetrialcomplete3:
         option("return to play menu");
         option("try again");
-        menuxoff = -25;
         menuyoff = 70;
         break;
     case Menu::gamecompletecontinue:
         option("return to play menu");
-        menuxoff = -25;
         menuyoff = 70;
         break;
     }
+
+    // Automatically center the menu. We must check the width of the menu with the initial horizontal spacing.
+    // If it doesn't fit onscreen (including square brackets), reduce the horizontal spacing by 5 and retry.
+    bool done_once = false;
+    int menuwidth = 0;
+    for (; !done_once || (menuwidth > 272 && menuspacing > 0); maxspacing -= 5)
+    {
+        done_once = true;
+        menuspacing = maxspacing;
+        menuwidth = 0;
+        for (size_t i = 0; i < menuoptions.size(); i++)
+        {
+            int width = i*menuspacing + graphics.len(menuoptions[i].text);
```

Comment:
The comment above says this counts square brackets, yet I don't see where in this line it accounts for them.

Reply:
Yeah that could have probably been clearer.

The menu is tried to be limited to 272 pixels (see the outer `for` loop). That is 320 (screen width) minus 16\*2 for brackets on each side, minus 8\*2 padding because it looks better to not extend to the borders of the screen. In the functions that draw menu options, the X positions of the menu options are where the text starts, when selected, the option including brackets are displayed 16 pixels more to the left.

Reply:
Yeah, makes sense.

Reply:
Let's document all that in this block, I was going to ask the same thing... other than that this seems fine to me, this will probably make life a lot easier if we ever have to add vertical menu scrolling.

Reply:
Done in 803d0f45de320d38cfe24a05595361764ebb5362.